x-pack/plugins/enterprise_search/public/applications/applications/components/search_application/search_preview/search_preview.tsx

```diff
@@ -407,7 +415,45 @@ export const SearchApplicationSearchPreview: React.FC = () => {
         <SearchProvider config={config}>
           <EuiFlexGroup>
             <EuiFlexItem>
-              <SearchBox inputView={InputView} />
+              <SearchBox
+                inputView={({ getInputProps }) => (
```

Comment:
suggestion: Make it more declarative and keep code in search_ui_components.

Example

```tsx
<SearchBox inputView={({ getInputProps }) => ( <SearchBar inputProps={getInputProps({
                        append: (
                          <EuiButtonEmpty
                            color="primary"
                            iconType="eye"
                            onClick={() => setShowAPICallFlyout(true)}
                            isLoading={lastAPICall == null}
                          >
                            {i18n.translate(
                              'xpack.enterpriseSearch.content.engine.searchPreview.inputView.appendButtonLabel',
                              { defaultMessage: 'View API call' }
                            )}
                          </EuiButtonEmpty>
                        ),
                      })} /> ) 
/>

const SearchBar = ({inputProps}) => (
      <EuiFlexGroup gutterSize="s">
                    <EuiFieldSearch
                      fullWidth
                      placeholder={i18n.translate(
                        'xpack.enterpriseSearch.content.engine.searchPreview.inputView.placeholder',
                        { defaultMessage: 'Search' }
                      )}
                      {...inputProps}
                  />
                       <EuiButton type="submit" color="primary" fill>
                      {i18n.translate(
                        'xpack.enterpriseSearch.content.engine.searchPreview.inputView.searchLabel',
                        { defaultMessage: 'Search' }
                      )}
                    </EuiButton>
  </EuiFlexGroup>
```

Reply:
This solution is really nice and much better 🚀  Thanks!

Reply:
Added changes in b02bfa554cca32e5f34a16741c33e439e64eedae